Tutorials/drawing-options/drawing-options-functions-using-bitmap.md

```diff
@@ -0,0 +1,176 @@
+
+   ```cpp
+   #include "spalshkit.h"
+   int main()
+    {
+       open_window("Drawing", 800, 600);
+
+       bitmap bmp = load_bitmap("my_bitmap", "bitmap.png");
+       drawing_options opts = option_part_bmp(0, 0, 100, 100);
+
+       draw_bitmap(bmp, 100, 100, opts);
+
+       delay(5000); // Pause for observation
+
+       close_all_windows();
+       return 0;
+    }
+
+## Example 2: Applying Option Rotate Bmp
+
+In this example, we'll showcase how to use 'Option Rotate Bmp' to dynamically rotate a bitmap at different angles.
+
+1. **Function Syntax:**
+
+ ```cpp
+  drawing_options option_rotate_bmp(double degrees);
+ ```
+
+- `degrees`: The angle of rotation in degrees.
+
+1. **Example Usage:**
+
+   In this code snippet, a window titled `Drawing` is initialized with dimensions of `800` pixels in width and `600` pixels in height using the open_window function. Next, a bitmap named `my_bitmap` is loaded from the file path `bitmap.png` using the load_bitmap function. Drawing options are applied to rotate the bitmap by `90 degrees clockwise` using the `option_rotate_bmp` function. Finally, the rotated bitmap is drawn onto the initialized window at coordinates (100, 100) using the `draw_bitmap` function.
+
+   ```cpp
+   #include "spalshkit.h"
+   int main()
+    {
+       open_window("Drawing", 800, 600);
+       
+       bitmap bmp = load_bitmap("my_bitmap", "bitmap.png");
+       drawing_options opts = option_rotate_bmp(90);
+
+       draw_bitmap(bmp, 100, 100, opts);
+
+       delay(5000);
+
+       close_all_windows();
+       return 0;
+    }
+
+## Example 3: Applying Option Scale Bmp
+
+In this example, we'll illustrate how to use 'Option Scale Bmp' to scale a bitmap to a desired size while maintaining its aspect ratio.
+
+1. **Function Syntax:**
+
+ ```cpp
+  drawing_options option_scale_bmp(double scale_x, double scale_y);
+ ```
+
+- `scale_x`: Scaling factor for the x-axis.
+  `scale_y`: Scaling factor for the y-axis.
+
+1. **Example Usage:**
+
+   In this code snippet, a window titled `Drawing` is initialized with dimensions of `800` pixels in width and `600` pixels in height using the open_window function. Next, a bitmap named `my_bitmap` is loaded from the file path `bitmap.png` using the load_bitmap function. Drawing options are applied to scale the bitmap to `150%` of its original size using the `option_scale_bmp` function. Finally, the scaled bitmap is drawn onto the initialized window at coordinates (100, 100) using the `draw_bitmap` function.
+
+   ```cpp
+   #include "spalshkit.h"
+   int main()
+    {
+       open_window("Drawing", 800, 600);
+       
+       bitmap bmp = load_bitmap("my_bitmap", "bitmap.png");
+       drawing_options opts = option_scale_bmp(1.5, 1.5);
+
+       draw_bitmap(bmp, 100, 100, opts);
+
+       delay(5000);
+
+       close_all_windows();
+       return 0;
+    }
+
+## Example 4: Combining Options
```

Comment:
This heading should be H3 (### instead of ##) as it is in the group of the heading above the first example section.